kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandler.java

```diff
@@ -476,34 +487,46 @@ public void channelInactive(ChannelHandlerContext ctx) {
         }
         final Channel outboundChannel = outboundCtx.channel();
         if (outboundChannel != null) {
-            closeNoResponse(outboundChannel);
+            closeWithNoResponse(outboundChannel);
         }
     }
 
     @Override
     public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
         LOGGER.warn("Netty caught exception from the frontend: {}", cause.getMessage(), cause);
-        if (cause instanceof DecoderException de && de.getCause() instanceof FrameOversizedException e) {
+        if (cause instanceof DecoderException de
+                && de.getCause() instanceof FrameOversizedException e) {
             var tlsHint = virtualCluster.getDownstreamSslContext().isPresent() ? "" : " or an unexpected TLS handshake";
             LOGGER.warn(
                     "Received over-sized frame, max frame size bytes {}, received frame size bytes {} "
                             + "(hint: are we decoding a Kafka frame, or something unexpected like an HTTP request{}?)",
                     e.getMaxFrameSizeBytes(), e.getReceivedFrameSizeBytes(), tlsHint);
         }
-        closeNoResponse(ctx.channel());
+        closeWithNoResponse(ctx.channel());
+    }
+
+    void closeInboundWithNoResponse() {
```

Comment:
I'd like to separate this part of the change as I think this is a good move and reduces blurring of boundaries between the frontend and backend handlers.